### PR TITLE
Composite measures

### PR DESCRIPTION
**Pull request by hackrm**
Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

Composite Measures can now be loaded.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1799
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Test fixtures updated and documented as necessary ( see [internal wiki](https://gitlab.mitre.org/bonnie/internal-documentation/wikis/testing#test-fixtures) )
- [x] Code coverage has not gone down and all code touched or added is covered. 
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 

Branch | Back End Coverage | Front End Coverage
-- | -- | --
master | 89.54% | image
composite_measures| 89.71% |  image

- [x] Automated regression test(s) pass: N/A

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint -N/A

**Reviewer 1:**

Name: @mayerm94 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass NA
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree NA

**Reviewer 2:**

Name: @hossenlopp 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass N/A
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree N/A

### CODE REVIEW

**Review by mayerm94 — APPROVED** [review]

File: app/controllers/measures_controller.rb [hunk 1]
```diff
@@ -220,6 +168,16 @@ def destroy
     end
     if cql_measure
       measure = cql_measure
+      if measure.component
```
Comment (hossenlopp) [inline app/controllers/measures_controller.rb:171]: A test should be made to test the contents of this if statement.

File: app/helpers/measure_helper.rb [hunk 2]
```diff
@@ -45,4 +45,78 @@ def self.build_vsac_error_message(e)
...
+
+      # Exclude patient birthdate and expired OIDs used by SimpleXML parser for AGE_AT handling and bad oid protection in missing VS check
+      missing_value_sets = (measure.as_hqmf_model.all_code_set_oids - measure.value_set_oids - ['2.16.840.1.113883.3.117.1.7.1.70', '2.16.840.1.113883.3.117.1.7.1.309'])
+      next if missing_value_sets.empty?
+      measures.each(&:delete)
+      error_message = {title: "Measure is missing value sets", summary: "The measure you have tried to load is missing value sets.", body: "The measure you are trying to load is missing value sets.  Try re-packaging and re-exporting the measure from the Measure Authoring Tool.  The following value sets are missing: [#{missing_value_sets.join(', ')}]"}
+      return error_message
+    end
+    if existing && is_update
+      existing.component_hqmf_set_ids.each do |component_hqmf_set_id|
```
Comment (hossenlopp) [inline app/helpers/measure_helper.rb:78]: The contents of this each block are not covered.

Comment (zlister) [reply app/helpers/measure_helper.rb:78]: May have to discuss this block further. Doesn't seem to get hit when updating a measure.

Comment (zlister) [reply app/helpers/measure_helper.rb:78]: Nevermind. I was missing a parameter.

File: test/functional/measures_controller_composite_test.rb [hunk 3]
```diff
@@ -166,7 +166,7 @@ class << measure_file
     
     assert_equal 8, CqlMeasure.all.count
   end
-  
+
```
Comment (jbradl11) [inline test/functional/measures_controller_composite_test.rb:169]: Is Github messing up the diff or is this the only change included in the commit? If so, I think changes are missing.

File: app/helpers/measure_helper.rb (hunk 2 above)
Comment (hossenlopp) [reply app/helpers/measure_helper.rb:78]: Still not seeing it covered with the latest commit.

File: test/functional/measures_controller_composite_test.rb [hunk 4]
```diff
@@ -207,4 +207,56 @@ class << measure_file
...
+        vsac_query_include_draft: 'false',
+        vsac_query_measure_defined: 'true',
+        vsac_username: ENV['VSAC_USERNAME'], vsac_password: ENV['VSAC_PASSWORD'],
+        measure_file: measure_file,
+        measure_type: 'ep',
+        calculation_type: 'patient',
+        hqmf_set_id: "244B4F52-C9CA-45AA-8BDB-2F005DA05BFC"
+      }
+    end
+    assert_response :redirect
```
Comment (hossenlopp) [inline test/functional/measures_controller_composite_test.rb:257]: This test should assert that the `id` of the composite has changed after the update. This ensures that it was actually changed out.

**Review by hossenlopp — APPROVED** [review]

**Pull request merged**